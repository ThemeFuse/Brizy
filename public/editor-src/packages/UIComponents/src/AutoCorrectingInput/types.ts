import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
