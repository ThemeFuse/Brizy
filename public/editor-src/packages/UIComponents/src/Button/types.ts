import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
