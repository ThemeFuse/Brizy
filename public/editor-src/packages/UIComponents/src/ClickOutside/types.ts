import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
