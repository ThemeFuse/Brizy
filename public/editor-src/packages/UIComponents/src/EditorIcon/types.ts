export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
