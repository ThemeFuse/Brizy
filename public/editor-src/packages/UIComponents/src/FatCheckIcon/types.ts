import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
