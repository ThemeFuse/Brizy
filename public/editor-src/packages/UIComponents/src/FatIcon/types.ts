import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
