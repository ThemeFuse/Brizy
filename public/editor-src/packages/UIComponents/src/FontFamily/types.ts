import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
