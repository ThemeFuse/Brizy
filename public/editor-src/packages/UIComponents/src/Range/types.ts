import { Handle } from "rc-slider";
import { ComponentProps, ReactNode, CSSProperties } from "react";

export interface Props {
  min?: number;
  max?: number;
  step?: number;
  startPointer: number;
  finishPointer: number;
  railStyle?: CSSProperties;
  handle?: (props: HandleProps) => ReactNode;
  onChange: (value: number[]) => void;
  className?: string;
}

export type HandleProps = ComponentProps<typeof Handle> & {
  index: number;
  dragging: boolean;
};
