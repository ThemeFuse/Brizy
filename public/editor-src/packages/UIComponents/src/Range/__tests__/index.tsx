import { fireEvent, render, screen } from "@testing-library/react";
import classNames from "classnames";
import { Handle } from "rc-slider";
import React, { ReactNode, useState } from "react";
import { Range } from "../index";
import { HandleProps } from "../types";

const MockComponent = () => {
  const [value, setValue] = useState({
    startPointer: 0,
    finishPointer: 100,
    active: "start"
  });

  const onStartChange = (startPointer: number): void => {
    setValue((prevState) => ({ ...prevState, startPointer }));
  };

  const onEndChange = (finishPointer: number): void => {
    setValue((prevState) => ({ ...prevState, finishPointer }));
  };

  const onChange = (v: number[]): void => {
    const c = v[0] === value.startPointer ? onEndChange : onStartChange;
    const _v: number = v[0] === value.startPointer ? v[1] : v[0];
    c(_v);
  };

  // only for custom classnames for thumbs when one of them is active
  const onActiveChange = (active: string): void => {
    setValue((prevState) => ({ ...prevState, active }));
  };

  // for custom classname when one of the thumbs is active
  const handle = (props: HandleProps): ReactNode => {
    const { className, index, ...other } = props;
    const _active = index === 0 ? "start" : "end";
    const _className = classNames(className, {
      "brz-ed-rc-slider-handle": true,
      "brz-ed-rc-slider-handle--active":
        (index === 0 && value.active === "start") ||
        (index === 1 && value.active === "end")
    });

    return (
      <Handle
        {...other}
        key={index}
        className={_className}
        // @ts-expect-error: missing props onMouseDown in ts
        onMouseDown={(): void => onActiveChange(_active)}
      />
    );
  };

  return (
    <Range
      min={0}
      max={100}
      step={1}
      handle={handle}
      startPointer={value.startPointer}
      finishPointer={value.finishPointer}
      onChange={onChange}
      className="custom-classname"
      railStyle={{
        backgroundImage: `linear-gradient(to right, #ebdb34, #3aeb34)`
      }}
    />
  );
};

describe("Test Range", () => {
  describe("Snapshots", () => {
    test("Render Range", () => {
      const { container } = render(<MockComponent />);

      expect(container).toMatchSnapshot();
    });

    test("Check onChange", () => {
      render(<MockComponent />);

      window.HTMLElement.prototype.getBoundingClientRect = () =>
        ({
          bottom: 44,
          height: 24,
          left: 10,
          right: 200,
          top: 20,
          width: 200
        } as DOMRect);

      const range = screen.getAllByRole("slider");

      expect(range[0]).toHaveClass("brz-ed-rc-slider-handle--active");

      fireEvent.mouseDown(range[1]);
      expect(range[1]).toHaveClass("brz-ed-rc-slider-handle--active");

      fireEvent.mouseDown(range[0], { clientX: 0, clientY: 0 });
      expect(range[0]).toHaveClass("brz-ed-rc-slider-handle--active");
    });
  });
});
