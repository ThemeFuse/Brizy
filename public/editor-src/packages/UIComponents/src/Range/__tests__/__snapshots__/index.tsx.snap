// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Range Snapshots Render Range 1`] = `
<div>
  <div
    class="brz-ed-control__range custom-classname"
  >
    <div
      class="rc-slider"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track rc-slider-track-1"
        style="left: 0%; width: 100%;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle rc-slider-handle-1 brz-ed-rc-slider-handle brz-ed-rc-slider-handle--active"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="100"
        class="rc-slider-handle rc-slider-handle-2 brz-ed-rc-slider-handle"
        role="slider"
        style="left: 100%;"
        tabindex="0"
      />
      <div
        class="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;
